pkg/types.go

```diff
@@ -0,0 +1,45 @@
+package pkg
```

Comment:
@twz123 I wanted to get your feedback about this convention within Go.

As I started to get into the refactoring of InjectRoute, I found that I kept getting into circular dependency type situations between the `bgp` and `routes` packages.

After watching a [video from Brian Ketelsen](https://www.youtube.com/watch?v=MzTcsI6tn-0) about structuring go programs, as well as several other posts, it seems like pulling up your domain models into the root of the project is a conventional way of working around circular dependencies.

Doing this with the below methods, did work out for me, but I noticed that this architecture is limited. For instance, I couldn't leave the `TunnelCreator` interface with `EncapType` and `EncapPort` because those are declared in the `tunnels` package. And for some reason, it didn't feel right to bring those up into this top level package either.

So I worked around it for now, but maybe this is showing that either I should have brought them up, or that maybe I'm embracing the wrong pattern here.

I would be curious for your thoughts.